aws/rust-runtime/aws-config/src/profile/parser.rs

```diff
@@ -153,6 +175,44 @@ impl ProfileSet {
         self.sso_sessions.get(name)
     }
 
+    /// Returns a list of all section names and their section type in tuple form e.g. `("profile", "user_a"), ("sso_session", "some_session"), ("services", "local_dev")`
+    ///
+    /// No sort order is guaranteed.
+    ///
+    /// # Example
+    ///
+    /// TODO add example
+    pub fn sections(&self) -> impl Iterator<Item = (&str, &str)> {
+        let other_sections = self
+            .other_sections
+            .iter()
+            .flat_map(|(k1, v)| v.keys().map(|k2| (k1.as_str(), k2.as_str())));
+        let profiles = self.profiles().map(|it| ("profile", it));
+        let sso_sessions = self.sso_sessions().map(|it| ("sso_session", it));
+
+        other_sections.chain(profiles).chain(sso_sessions)
+    }
+
+    /// Given a section type and name, retrieve the corresponding section, if it exists.
+    ///
+    /// # Example
+    ///
+    /// TODO Make this work for profile and sso_session
+    pub fn get_raw_section_with_properties<'a>(
+        &'a self,
+        section_type: &str,
+        section_name: &str,
+    ) -> Option<&HashMap<String, HashMap<String, String>>> {
```

Comment:
I'm slightly concerned enforcing a pairing of section_type/section_name in the public API could cause issues in the future given we have sections such as `[default]`.

Reply:
In the case of `[default]`, I see that as just an unnamed `profile`-type section. I'm willing to budge if we really thing that we'll have 'typeless sections' in the future though.